### PR TITLE
Add the missing logic to handle pre-created ClusterBootstrap

### LINKED ISSUE #2603: ConfigAPI objects are not cloned so it throws an error
**Bug description**
The ConfigAPI objects are not cloned so it throws an error. For example, Kappconfig
Error:
`
The ClusterBootstrap "abc124" is invalid: spec.kapp.valuesFrom.ProviderRef: Invalid value: v1.TypedLocalObjectReference{APIGroup:(*string)(0xc0018a1e20), Kind:"KappControllerConfig", Name:"v1.23.5---vmware.1-tkg.1"}: kappcontrollerconfigs.run.tanzu.vmware.com "v1.23.5---vmware.1-tkg.1" not found`

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by danniel1205**
### What this PR does / why we need it

This PR contains the potential fixes for three GH issues: #2603, #2602, #2585.

Summary of the problem:
- When a ClusterBootstrap CR has been created by third part (e.g. Tanzu CLI), validation webhook complains the `valuesFrom` not found. That is because the corresponding CB packages have not yet been cloned (this is done by clusterbootstrap_controller)
- When a ClusterBootstrap CR has been created by third part (e.g. Tanzu CLI), and the `valuesFrom` fields are missing inside of `AdditionalPackages`. They are not added based off ClusterBootstrapTemplate.Spec.AdditionalPackages.
- When a ClusterBootstrap CR has been created by third part (e.g. Tanzu CLI), the `providerRef.Name` does not contain the cluster name.

This PR address all above problems.

### Which issue(s) this PR fixes

Fixes #2603 #2602 #2585

### Describe testing done for PR

- local unit tests and envtest passed
- Manually tested against a real cluster

[asciicast](https://asciinema.org/a/504838)

### Release note

```release-note

```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go
Comment (maralavi) [inline addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go:290]: Question: Should it be checked if this tkr actually exists?

File: addons/webhooks/clusterbootstrap_webhook_test.go
Comment (maralavi) [inline addons/webhooks/clusterbootstrap_webhook_test.go:125]: Question: In the case of this example, ```ClusterBootstrapTemplate``` had the same number of entries as ``` ClusterBootstrap ```?

Comment (danniel1205) [reply addons/webhooks/clusterbootstrap_webhook_test.go:126]: I will change it to `2` to make it readable and understandable.

File: addons/webhooks/clusterbootstrap_webhook_test.go
Comment (danniel1205) [reply addons/webhooks/clusterbootstrap_webhook_test.go:129]: good catch

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go
Comment (danniel1205) [reply addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go:305]: In few lines below, it checks the CBT (all it cares about is the presence of CBT )

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go
Comment (vijaykatam) [inline addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go:217]: Question: Would it make this logic simpler if we we did not have to merge additionalPackages? We would only handle copying additionalPackages if key did not exist in destination.

Note: Don't need to make the change here, lets just log issue if we need to simplify this.

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go
Comment (vijaykatam) [inline addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go:404]: Question: Should the annotation (clusterBootstrap.Annotations[constants.AddCBMissingFieldsAnnotationKey]) be removed as well?

File: addons/controllers/clusterbootstrap_controller.go
Comment (vijaykatam) [inline addons/controllers/clusterbootstrap_controller.go:395]: Also check that the add missing fields from annotation exists.

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone_test.go
Comment (vijaykatam) [inline addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone_test.go:545]: Not sure how this test is different from above. This test is asserting additionalPackages which is covered above.

File: addons/webhooks/clusterbootstrap_webhook.go
Comment (vijaykatam) [inline addons/webhooks/clusterbootstrap_webhook.go:94]: Question: What use case would trigger this? Default is supposed to be called on create only?

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go
Comment (vijaykatam) [inline addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go:195]: Since we don't support this in our data model, I would suggest log a warning(if not error) and don't do anything.

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go
Comment (vijaykatam) [inline addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go:185]: go has copy() for slices

**Review by vijaykatam — APPROVED** [review]
Thanks for the changes. Nothing that is blocking. We can address as follow ups. Gonna approve as TKGS team is blocked on testing.

**Comment by danniel1205** [comment]
In order to unblock others from testing, I would merge this PR first. Discussed with @vijaykatam that we agreed to address those non-blocking comments  in a separate transaction.

**Pull request merged**
